### DISCUSSION TITLE
Source tree re-org

### DISCUSSION THREAD

**Discussion by teverett** [antlr/grammars-v4#2871] [General]
It's likely time to re-org the source tree to make it a little more manageable.   The timing to do this would be "after we have completed the work to generate a human-readable indecx", something which @kaby76 and @parrt  are working on.

This schema has been proposed by @KOLANICH, [here](https://github.com/antlr/grammars-v4/issues/2826)

* `config` - config files and records.
* `grammar` - DSLs describing other grammars.
  * `text` - grammars like the ones for tools like ANTLR
  * `ddl` - DSLs for describing binary grammars, like `protobuf`, `flatbuffers`, `capnproto`, `FlexT` and so on
* `programming` - programming and scripting languages, like C++ or bash.
* `programms` - for parsing output of software, when it is infeasible to use a machine-readable interface.
* `protocols` - for interfacing servers or devices, single command per line, such as SCPI, AT, JTAG consoles, SMTP, stuff like this.
* `serialization` - serialization languages, like JSON, YAML, protobuf and CSV.
* `embedded` - grammars used as parts of other formats, that don't belong to anywhere else
* `identifiers` - various identifiers, like SSNs, phone numbers, VIN-codes, UUID and so on
  * `network` - network addresses: IPv4, IPv6, MAC, IMEI,
  * `products` - product namebers, like [`HTE721010A9E630`](https://github.com/KOLANICH-ML/HDDModelDecoder.py)

The rest of identifiers should stay in `root` untill it is decided to where they are to be moved.

Are there any opinions or perspective on the schema to adopt for a source tree re-org?

**Comment by parrt** [comment]
Makes sense to get a good organization, but is it the file directories structure that needs to be organized that way or just multiple indexes that provide different ontologies?

**Comment by teverett** [comment]
Could be either or both.   It's worth noting that there is already some file system hierarchy such as /asm.  My concern with it is that to a casual browser of the GH main pages for grammars-v4 they may not notice that asm/asm8080 exists.  Alternatively, putting everything in the root makes for quite a lot of grammars in the root.  The schema @KOLANICH suggested, combined with a full index merged into readme.md provides a nice balance.

**Comment by KvanTTT** [comment]
I suggest moving the thread to another discussion that was converted from the similar issue: https://github.com/antlr/grammars-v4/discussions/2872
